From 40739dde552e648863f61aeb85d41a6e6ad3b683 Mon Sep 17 00:00:00 2001
From: Tiffany Kalin <tkalin@untangle.com>
Date: Mon, 6 Jul 2020 14:42:03 -0600
Subject: [PATCH] Add in logic to determine architecture when building linux in
 openwrt

In order to build libbpf, the right includes need to be found and this depends on the architecture. The usual linux tools Makefiles don't know what architecture openwrt is currently being build in, so we use the CC variable that has the ARCH embedded in it in order to populate architecture variables so the right device and includes are built for libbpf.
---
 tools/scripts/Makefile.arch    | 10 ++++++++--
 tools/scripts/Makefile.openwrt | 12 ++++++++++++
 2 files changed, 20 insertions(+), 2 deletions(-)
 create mode 100644 tools/scripts/Makefile.openwrt

diff --git a/tools/scripts/Makefile.arch b/tools/scripts/Makefile.arch
index 78d90a249e88..eb152f8bdd36 100644
--- a/tools/scripts/Makefile.arch
+++ b/tools/scripts/Makefile.arch
@@ -7,8 +7,14 @@ HOSTARCH := $(shell uname -m | sed -e s/i.86/x86/ -e s/x86_64/x86/ \
                                   -e s/sh[234].*/sh/ -e s/aarch64.*/arm64/ \
                                   -e s/tile.*/tile/ )
 
-ifndef ARCH
-ARCH := $(HOSTARCH)
+ifneq (,$(findstring openwrt,$(CC)))
+  ifndef ARCH
+    include $(srctree)/tools/scripts/Makefile.openwrt
+  endif 
+else 
+  ifndef ARCH
+    ARCH := $(HOSTARCH)
+  endif
 endif
 
 SRCARCH := $(ARCH)
diff --git a/tools/scripts/Makefile.openwrt b/tools/scripts/Makefile.openwrt
new file mode 100644
index 000000000000..c72f72938a1c
--- /dev/null
+++ b/tools/scripts/Makefile.openwrt
@@ -0,0 +1,12 @@
+# SPDX-License-Identifier: GPL-2.0
+
+ifneq (,$(findstring x86_64,$(CC)))
+  ARCH := x86
+endif
+ifneq (,$(findstring aarch64,$(CC)))
+  ARCH := aarch64
+endif
+
+ifndef ARCH
+$(error Arch not supported for libbpf)
+endif
-- 
2.20.1

